packages/@aws-cdk/aws-pipes-alpha/lib/logs.ts

```diff
@@ -93,3 +153,81 @@ export interface ILogDestination {
    */
   grantPush(grantee: IRole): void;
 }
+
+/**
+ * CloudWatch Logs log group for delivery of pipe logs.
+ */
+export class CloudwatchLogsLogDestination implements ILogDestination {
+  private logGroup: ILogGroup;
+
+  constructor(logGroup: ILogGroup) {
+    this.logGroup = logGroup;
+  }
+
+  bind(_pipe: IPipe): LogDestinationConfig {
+    return {
+      parameters: {
+        cloudwatchLogsLogDestination: {
+          logGroupArn: this.logGroup.logGroupArn,
+        },
+      },
+    };
+  }
+
+  grantPush(pipeRole: IRole): void {
+    this.logGroup.grantWrite(pipeRole);
+  }
+}
+
+/**
+ * Firehose stream for delivery of pipe logs.
+ */
+export class FirehoseLogDestination implements ILogDestination {
+  private deliveryStream: IDeliveryStream;
+
+  constructor(deliveryStream: IDeliveryStream) {
+    this.deliveryStream = deliveryStream;
+  }
+
+  bind(_pipe: IPipe): LogDestinationConfig {
+    return {
+      parameters: {
+        firehoseLogDestination: {
+          deliveryStreamArn: this.deliveryStream.deliveryStreamArn,
+        },
+      },
+    };
+  }
+
+  grantPush(pipeRole: IRole): void {
+    this.deliveryStream.grantPutRecords(pipeRole);
+  }
+}
+
+/**
+ * S3 bucket for delivery of pipe logs.
+ */
+export class S3LogDestination implements ILogDestination {
+  private parameters: S3LogDestinationProps;
+
+  constructor(parameters: S3LogDestinationProps) {
+    this.parameters = parameters;
+  }
+
+  bind(_pipe: IPipe): LogDestinationConfig {
+    return {
+      parameters: {
+        s3LogDestination: {
+          bucketName: this.parameters.bucket.bucketName,
+          bucketOwner: this.parameters.bucketOwner || this.parameters.bucket.env.account,
+          outputFormat: this.parameters.outputFormat,
+          prefix: this.parameters.prefix,
+        },
+      },
+    };
+  }
+
+  grantPush(pipeRole: IRole): void {
+    this.parameters.bucket.grantWrite(pipeRole);
+  }
```

Comment:
Is this the least privileged principal we can grant? Are we able to get away with using a less-privileged permission like `bucket.grantPut()` instead?